logprep/processor/pre_detector/processor.py

```diff
@@ -103,7 +107,20 @@ def process(self, event: dict) -> tuple:
         self._event = event
         detection_results = []
 
-        for rule in self._tree.get_matching_rules(event):
+        for rule in self._generic_tree.get_matching_rules(event):
+            begin = time()
+            if not (
+                self._ip_alerter.has_ip_fields(rule)
+                and not self._ip_alerter.is_in_alerts_list(rule, event)
+            ):
+                if self._logger.isEnabledFor(DEBUG):
+                    self._logger.debug("{} processing matching event".format(self.describe()))
+                self._get_detection_result(rule, detection_results)
+                processing_time = float("{:.10f}".format(time() - begin))
```

Comment:
Can you use a f-string here to comply with the linting?

Reply:
I have no guess, why it is written this way cause if you write it this way no test is failing
```
processing_time = time() - begin
```
that is what I implemented now